### PR TITLE
[Merged by Bors] - feat: port Topology.ShrinkingLemma

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/ShrinkingLemma.lean [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+      use j
+      have hj' : x ∈ u j := (v i).subset _ hj
+      have : v j ≤ v i := (hc.total (hvc _ hxi) (hvc _ hj')).elim (fun h => (hmax j hj' h).ge) id
+      simpa only [find_apply_of_mem hc ne (hvc _ hxi) (this.1 <| hiv _ hj')]
+  closure_subset hi := (find c ne _).closure_subset ((mem_find_carrier_iff _).2 hi)
+  apply_eq hi := (find c ne _).apply_eq (mt (mem_find_carrier_iff _).1 hi)
+#align shrinking_lemma.partial_refinement.chain_Sup ShrinkingLemma.PartialRefinement.chainSup
+
+/-- `chain_Sup hu c hc ne hfin hU` is an upper bound of the chain `c`. -/
+theorem le_chainSup {c : Set (PartialRefinement u s)} (hc : IsChain (· ≤ ·) c) (ne : c.Nonempty)
```
Comment (dupuisf) [inline Mathlib/Topology/ShrinkingLemma.lean:169]:
```suggestion
/-- `chainSup hu c hc ne hfin hU` is an upper bound of the chain `c`. -/
theorem le_chainSup {c : Set (PartialRefinement u s)} (hc : IsChain (· ≤ ·) c) (ne : c.Nonempty)
```

File: Mathlib/Topology/ShrinkingLemma.lean [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+noncomputable section
+
+variable {ι X : Type _} [TopologicalSpace X] [NormalSpace X]
+
+namespace ShrinkingLemma
+
+-- the trivial refinement needs `u` to be a covering
+/-- Auxiliary definition for the proof of `shrinking_lemma`. A partial refinement of a covering
+`⋃ i, u i` of a set `s` is a map `v : ι → Set X` and a set `carrier : Set ι` such that
```
Comment (dupuisf) [inline Mathlib/Topology/ShrinkingLemma.lean:47]: Does `shrinking_lemma` actually exist?

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

File: Mathlib/Topology/ShrinkingLemma.lean [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+open scoped Classical
+
+noncomputable section
+
+variable {ι X : Type _} [TopologicalSpace X] [NormalSpace X]
+
+namespace ShrinkingLemma
+
+-- the trivial refinement needs `u` to be a covering
+/-- Auxiliary definition for the proof of `shrinking_lemma`. A partial refinement of a covering
```
Comment (urkud) [inline Mathlib/Topology/ShrinkingLemma.lean:46]:
```suggestion
/-- Auxiliary definition for the proof of the shrinking lemma. A partial refinement of a covering
```

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**